### ISSUE TITLE
De/serializing DateTime and Duration

### ISSUE THREAD

**Issue by nrempel** [tafia/quick-xml#339]
Hi there,

We have a need to serialize and deserialize durations and dates/times as defined in the [XML Schema spec](https://www.w3.org/TR/xmlschema-2/#duration).

Do you think this functionality would fit as part of this project or should best be represented in a separate crate?

**Comment by Mingun** [comment]
I think that we should support dates and times according to XML Schema spec. PR is welcome!

**Comment by nrempel** [comment]
Awesome, thank you! We'll look at this again soon.

**Comment by Mingun** [comment]
Because serde has no built-in support for dates, that is unclear what support we should provide. It could be a function / module for `#[serde(with = "...")]`, but it seems that these functions [already provided](https://docs.rs/chrono/latest/chrono/serde/index.html) in `chrono`, but if not it is probably better to include them there.

**Comment by djc** [comment]
I actually think this is already somewhat working? That is, it gets used for example:

https://github.com/masalachai/epp-client/blob/main/src/domain/info.rs#L107

Don't quite remember why we (@nrempel is my co-worker) filed this in the first place.

**Comment by nrempel** [comment]
Yeah I think I was referring to durations specifically as defined by the xml spec.

We have custom serialization for that here: https://github.com/masalachai/epp-client/blob/main/src/extensions/consolidate.rs
